src/validationinterface.cpp

```diff
@@ -5,9 +5,13 @@
 
 #include <validationinterface.h>
 
+#include <chain.h>
+#include <logging.h>
 #include <primitives/block.h>
+#include <primitives/transaction.h>
```

Comment:
From the [Developer Notes - Source code organization](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#source-code-organization):
> - Every `.cpp` and `.h` file should `#include` every header file it directly uses classes, functions or other definitions from, even if those headers are already included indirectly through other headers.
>    -  _Rationale_: Excluding headers because they are already indirectly included results in compilation failures when those indirect dependencies change. Furthermore, it obscures what the real code dependencies are.

Reply:
Ah thanks, didn't know it!